src/submission/validation-clinical/therapy.ts

```diff
@@ -17,42 +17,234 @@
+  let errors: SubmissionValidationError[] = [];
   const treatment = getTreatment(therapyRecord, mergedDonor, errors);
   if (!treatment) return { errors };
-  checkTreatementHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+  // TODO: Refactor use of push
+  checkTreatmentHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+
+  const crossFileErrors = await crossFileValidator(
+    mergedDonor,
+    treatment,
+    therapyRecord,
+    submittedRecords,
+  );
+
+  errors = [...errors, ...crossFileErrors];
   return { errors };
 };
 
-export function checkTreatementHasCorrectTypeForTherapy(
+const crossFileValidator = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  // submitted records needed to validate current submission
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const isRadiationRecord = Object.values(RadiationFieldsEnum).some(field =>
+    Object.keys(therapyRecord).includes(field),
+  );
+
+  const radiationErrors: SubmissionValidationError[] = isRadiationRecord
+    ? await validateRadiationRecords(donor, treatment, therapyRecord, submittedRecords)
+    : [];
+
+  const crossFileErrors = [...radiationErrors];
+
+  return crossFileErrors;
+};
+
+interface DonorResult {
+  donors: DeepReadonly<Donor>[];
+}
+
+const emptyDonorResult: DonorResult = { donors: [] };
+
+const validateRadiationRecords = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const { programId } = donor;
+
+  const {
+    clinicalInfo: {
+      submitter_donor_id: relatedTreatmentDonorId,
+      submitter_treatment_id,
+      treatment_type,
+    },
+  } = treatment;
+
+  const {
+    submitter_donor_id: submittedTherapyDonorId,
+    radiation_boost,
+    reference_radiation_treatment_id,
+  } = therapyRecord;
+
+  // Compare across other Submissions
+  let submittedTreatments: DeepReadonly<ClinicalInfo>[] = [];
+
+  for (const donorId in submittedRecords) {
+    const submittedTreatmentRecords = [...submittedRecords[donorId].treatment] as ClinicalInfo[];
+    if (submittedTreatmentRecords && submittedTreatmentRecords.length)
+      submittedTreatments = [...submittedTreatments, ...submittedTreatmentRecords];
+  }
+
+  const query: ClinicalQuery = {
+    programShortName: programId,
+    entityTypes: [ClinicalEntitySchemaNames.TREATMENT, ClinicalEntitySchemaNames.RADIATION],
+    page: 0,
+    sort: 'donorId',
+    donorIds: [],
+    submitterDonorIds: [],
+  };
+
+  // Compare across all Treatments
+  const { donors } =
+    (await donorDao.findByPaginatedProgramId(programId, query)) || emptyDonorResult;
+
+  const storedTreatments = donors.length
+    ? donors
+        .map(donor => donor.treatments)
+        .filter(treatments => Boolean(treatments))
+        .flat()
+    : [];
```

Comment:
you can save an iteration of the array by just using a `filter`
most array methods will give you back empty arrays if operating on empty array 
ie. they won't throw errors for empty arrays
`[].filter(x => x?.y?.z).flat() // result is []`


I think this could be simplified to:
```typescript
  const storedTreatments = donors.filter(donor => Boolean(donor?.treatments)).flat();
```

Reply:
Updated